### DISCUSSION TITLE
[Bug]: Utilização do componente Money em live() com Placeholder

### DISCUSSION THREAD

**Discussion by mliell** [leandrocfe/filament-ptbr-form-fields#53] [Q&A]
### What happened?

Ao utilizar o componente Money e obter um get em um Placeholder o mesmo está retornando valores incorretos.

Segue vídeo demonstrando em uma aplicação teste:

[Gravação de tela de 2023-08-31 14-42-04.webm](https://github.com/leandrocfe/filament-ptbr-form-fields/assets/4391791/282f0002-83aa-418a-98c6-90a0c8914e56)

Como podem ver no vídeo, somente obtendo um get do live() que os valores retornados estão incorretos, ao salvar no banco está ok.

Segue o repositório  do código utilizado: https://github.com/mliell/Erro-money-placeholder

### How to reproduce the bug

Crie um input tipo Money com live():
`Money::make('valor1')->live(),`

Retorne os dados em um placeholder:

`Placeholder::make('Valor_1')->content(function ($get) {
                        return $get('valor1');
                    }),`

Ou baixe o repósitório https://github.com/mliell/Erro-money-placeholder

### Package Version

3.0.1

### PHP Version

8.1.12

### Laravel Version

10.21.0

### Notes

Banco de dados utilizado: sqlite

**Comment by leandrocfe** [comment]
Testou `live(onBlur:true)`?

**Comment by mliell** [comment]
Sim, retorna o mesmo.

**Comment by mliell** [comment]
Um detalhe interessante. Se colocar o cursor no lado esquerdo do valor (ficando a esquerda da vírgula, sendo os números inteiros) e adicionar um número, o retorno é correto.
Agora se adicionar os números pelo lado esquerdo (decimal) que é o default, acontece o bug.

**Comment by leandrocfe** [comment]
é por conta do javascript..

**Comment by leandrocfe** [comment]
@mliell verifica essa versão por favor
https://github.com/leandrocfe/filament-ptbr-form-fields/releases/tag/3.1.0

**Comment by leoblanski** [comment]
Opa Leandro!!

Testei aqui mas o problema persiste...

leandrocfe/filament-ptbr-form-fields: 3.1.0

Exemplo:




https://github.com/user-attachments/assets/0ba7bdf1-f760-48c2-a5a7-74d94a9f95ca

**Comment by jefpes** [comment]
Esse problema acontece quando o component está ->live(), se vc tiver precisando fazer cálculos com os valores do component vc tem que fazer um verdadeiro giro pra funcionar de maneira adequada, para usar em uma resource de venda que exige cálculos com qtds, juros e etc, tive que fazer uma mudança enorme para funcionar de maneira adequada.

![image](https://github.com/user-attachments/assets/497fb877-c399-4990-9247-44161eec32b4)
![image](https://github.com/user-attachments/assets/75733390-7e5e-4614-8422-e758bbe43d85)
![image](https://github.com/user-attachments/assets/5fd0bdba-f7fe-4d3b-bb40-b054ffd02f7c)

nisso apesar de não ser o cenário ideal, vai resolver o problema com o ->live()

claro que vc deve adaptar para o seu cenário, até a correção do plugin.
